### PR TITLE
[main < T469-FL] Fix reset, class label and early stopping bugs on node classification module

### PR DESCRIPTION
**Pull request by antoniofilipovic**
### Description

This PR includes **three** bugfixes:
- training throws an error after calling the **`reset`** method
- training not working if **class labels don't start with 0**
- potential bug for **early stopping on the first iteration if val_loss > 100**

### Pull request type

- [x] Bugfix

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)

######################################

### CODE REVIEW

File: python/mage/node_classification/models/train_model.py [hunk 1]
```diff
@@ -70,7 +70,8 @@ def training_loop(loader: HGTLoader, gradient: bool) -> float:
             out = model(batch.x_dict, batch.edge_index_dict)[
                 observed_attribute
             ]  # Perform a single forward pass.
-
+            print(out) 
+            print(batch[observed_attribute].y)
```
Comment (niko4299) [inline python/mage/node_classification/models/train_model.py:74]: Don't forget to remove these 2 prints.

**Review by niko4299 — APPROVED** [review]

File: python/node_classification.py [hunk 2]
```diff
@@ -731,7 +733,12 @@ def predict(
 
     predicted_class = int(pred.detach().cpu().numpy()[position])
 
-    return mgp.Record(predicted_class=predicted_class, status="Prediction done.")
+    return mgp.Record(
+        predicted_class=current_values[HeteroParams.INV_LABEL_REINDEXING][
+            predicted_class
+        ],
+        status="Prediction done.",
```
Comment (antepusic) [inline python/node_classification.py:740]: Sounds better :slightly_smiling_face: 
```suggestion
        status="Prediction complete.",
```

**Review by antepusic — COMMENTED** [review]
All looks good so far :rocket:

**Review by antepusic — APPROVED** [review]
